### ISSUE TITLE
How to create a "Installed by default" plugin?

### ISSUE THREAD

**Issue by Leonardo-Ferreira** [grandnode/grandnode#646]
I'm working on a idea of a plugin to redirect Log messages to Application insights. 
I would like to have it installed by default and but not activated. 
I just can't find the place where we register the plugins...

Can anyone help?

**Comment by PatrykPorabik** [comment]
Hi,

Few types of plugins are disabled by default. For example if you install widgets or payment methods, you will get inactivate plugin.

**Comment by Leonardo-Ferreira** [comment]
Tks for the reply @PatrykPorabik but I don't think I made myself clear.
What I did was:

1. Create a new lib project on the Plugins folder, following the example of other plugins
2. Create a *Registrar file and the implementation of ILogger interface
3. Created the description.txt file accordantly
(technically, the plugin is just this)
4. Inserted it on the InstalledPlugins.txt at Grand.Web project
5. Started up the project

I was hoping to go to the admin section and see the plugin there! but it didn't show up... what extra steps should I take?

**Comment by huyhuana** [comment]
Did you mean that your Plugin did not show up in the /admin/plugin/list ?
In this case you need to build your Plugin, then , if it create a *.dll in the /Grand.Web/Plugins/ , your Plugin will show up in admin panel.

**Comment by marbriganti** [comment]
Hi, Could you to solve the problem? I'm having the same problem.

**Comment by Leonardo-Ferreira** [comment]
@marbriganti no... not so far... I actually decided to do that later, so I kinda "stashed" the issue

**Comment by PatrykPorabik** [comment]
I don't know how can I help you. But maybe I didn't understand the problem. In GrandNode if you created plugin correctly, when you Build it, it will be automatically copied to ~/Grand.Web/Plugins folder. Then you should go to Admin panel of your store, go Configuration -> Local plugins and install it from the list. You can't add it manually to InstalledPlugins.txt.

**Comment by Leonardo-Ferreira** [comment]
Perhaps the problem is how to create a plugin correctly. I for one, followed the steps described above and it didn't work as expected... perhaps you can point out what I did wrong. I did once tried without having it manually placed in the InstalledPlugins.txt

**Comment by marbriganti** [comment]
Hi @Leonardo-Ferreira 
I was able to solve the problem. You have to run the site without debugging, and go to Local Plugins, click on "Reload list of Plugins" and then you will see it in the list available to install. Regards!
